SonarQube.Common/AnalysisProperties/AnalysisPropertyFileProvider.cs

```diff
@@ -0,0 +1,150 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AnalysisPropertyFileProvider.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SonarQube.Common
+{
+    /// <summary>
+    /// Handles locating an analysis properties file and returning the appropriate properties
+    /// </summary>
+    public class AnalysisPropertyFileProvider : IAnalysisPropertyProvider
```

Comment:
There are currently two property providers. This one handles locating and getting settings from the new global settings file (i.e. looks for the file specified by the /s: parameter if there is one, failing which it looks for a default settings file).
The other handles validating and turning /p: settings from the command line into properties.

The aggregating provider handles combining the properties from the various providers correctly.